### DISCUSSION TITLE
Poco::FileStream std::fstream

### DISCUSSION THREAD

**Discussion by micheleselea** [pocoproject/poco#4565] [General]
Hi all, I recently done some tests about speed of writing file: in my case is a write to a samba share folder, but it's quite the same on a local drive. I found that if I write 24MB of data file with Poco::FileStream it takes about 2seconds, if I try with std::fstream it takes about 200 milliseconds
I think the reason is the "buffer class"  between the FileStream  and the WriteFile to disk (I'm using windows for these test), because it writes 4096 byte at a time (FileStreamBuf::BUFFER_SIZE). I think we already discuss here on an old post about it, but I don't remember what was the solution.
I think probably because std::fstream is c++ portable there's no reason to use Poco::FileStream or do you think that in some specific environment will be better using Poco::FileStream

**Comment by micheleselea** [comment]
Probably I realized that the Poco::FileStream has a better handle of error exceptions

**Comment by aleks-f** [comment]
https://github.com/pocoproject/poco/issues/2475

**Comment by aleks-f** [comment]
I think the main reason for FileStream was to handle unicode file names

**Comment by micheleselea** [comment]
I changed the code to insert this constructor
`FileStream(const std::string& path, std::ios::openmode mode = std::ios::out | std::ios::in, int BUFFER_SIZE = FileStreamBuf::BUFFER_SIZE);`
If I use like this
```
const int bufferFIleStreamSize = 512 * 1024;
Poco::FileStream fFile(sFileName, mode, bufferFIleStreamSize);
```
the performance are comparable to std::fstream

Obviously I had to change even FileIOS as well

**Comment by matejk** [comment]
`FileStreamBuf`has function `resizeBuffer` that can be used to set own size of the file stream buffer. Use `FileStream::rdbuf()` to acquire the stream buffer.

**Comment by micheleselea** [comment]
I don't see resizeBuffer in my source code, from which version is there that function?

**Reply by matejk** [reply]
It was introduced in version 1.14.
